contracts-periphery/src/UniswapWithdrawHook.sol

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+pragma abicoder v2;
+
+import "openzeppelin-contracts/contracts/access/Ownable.sol";
+import "openzeppelin-contracts/contracts/token/ERC20/utils/SafeERC20.sol";
+import "src/interfaces/ISwapRouter.sol";
+
+contract UniswapWithdrawHook is Ownable {
+  using SafeERC20 for IERC20;
+
+  ISwapRouter internal immutable swapRouter;
+  address internal WETH9;
+
+  uint256 public feeBips;
+  address public feeCollector;
+  address payable public feeReceiver;
+
+  constructor(ISwapRouter _swapRouter, address _WETH9, uint256 _feeBips, address payable _feeReceiver) {
+  swapRouter = _swapRouter;
+  WETH9 = _WETH9;
+  feeBips = _feeBips;
```

Comment:
Why not make the fee an input to each call, like Uniswap's router does, and just hardcode a max fee? That way we can more easily do things like:
- Change the fee without deploying a new contract
- Change fee by use case/token/for some time like a promotion or something

Reply:
Contract should now be much different https://github.com/ScopeLift/umbra-protocol/pull/341/commits/b40d4a8da174dc150d24e62c76a20189c2d8cfc2